### DISCUSSION TITLE
loadBalanceSegmentLoop: there are too few available query nodes to balance

### DISCUSSION THREAD

**Discussion by liuyichen2013** [milvus-io/milvus#18136] [Q&A and General discussion]
version:2.02
插入1.5亿条128维向量，会出现下面的提示，查询时milvus会重启。插入1亿条向量时，可以正常查询。请问是什么原因，谢谢！

[DEBUG] [query_coord.go:516] ["loadBalanceSegmentLoop: memory usage rate of all online QueryNode"] ["mem rate"="{\"21\":1.273337169038769e-8}"]
[2022/07/07 06:33:40.072 +00:00] [WARN] [query_coord.go:518] ["loadBalanceSegmentLoop: there are too few available query nodes to balance"] [onlineNodeIDs="[21]"] [availableNodeIDs="[21]"]

**Comment by liuyichen2013** [comment]
libgomp: Thread creation failed: Resource temporarily unavailable

libgomp: Thread creation failed: Resource temporarily unavailable

libgomp: Thread creation failed: Resource temporarily unavailable

libgomp: Thread creation failed: Resource temporarily unavailable

libgomp: Thread creation failed: Resource temporarily unavailable

libgomp: Thread creation failed: Resource temporarily unavailable

**Reply by xiaofan-luan** [reply]
libgomp: Thread creation failed: Resource temporarily unavailable 

It's usually a issue when you have too many threads.
1) pls check your ulimit setup, maybe set to 4096 will solve the problem
2) if we  can save pprof and pstack to check the process stats that could be really helpful
